assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils.ts

```diff
@@ -29,3 +35,55 @@ export const prepareEntitiesForUpdate = <Entity extends TAMRelationEntity>({
 		return !equals(newRelatedEntities.sort(), oldRelatedEntities.sort());
 	}, Object.entries(newEntities));
 };
+
+/**
+ * e.g
+ * {
+ *     2019: {
+ *         9: 'October',
+ *         10: 'Noember',
+ *         11: 'December',
+ *     },
+ *     2020: {
+ *         0: 'Januray',
+ *         1: 'February',
+ *         2: 'March',
+ *         3: 'April',
+ *     },
+ * }
+ */
+type YearWiseMonths = AnyObject<AnyObject<string>>;
+
+export const getMonthsListFromDatetimes = (dates: Array<Datetime>): OptionsType => {
+	const sortedDates = sortDates({ dates });
+
+	const yearWiseMonths = sortedDates.reduce<YearWiseMonths>((acc, { startDate }) => {
+		const parseDate = parseISO(startDate);
+		const year = parseDate.getFullYear();
+		const month = parseDate.getMonth();
+
+		const monthsInTheYear = acc[year] || {};
+		if (!(month in monthsInTheYear)) {
+			monthsInTheYear[month] = monthNames[month];
+
+			acc[year] = monthsInTheYear;
+		}
+		return acc;
+	}, {});
+
+	const list = Object.entries(yearWiseMonths).map(([year, months]) => {
+		return {
+			key: year,
+			label: year,
+			options: Object.entries(months).map(([number, name]) => {
+				return {
+					key: `${year}:${number}`,
+					label: name,
+					value: `${year}:${number}`,
+				};
+			}),
+		};
+	});
+
+	return list;
+};
```

Comment:
This can goto `shared` domain and can be used to add month filter to dates `EntityList`